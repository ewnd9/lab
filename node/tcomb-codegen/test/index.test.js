'use strict';

const gen = require('../src/');

it('string', () => {
  const code = gen('string');
  expect(code).toMatchSnapshot();
});

it('number', () => {
  const codeInteger = gen(1);
  expect(codeInteger).toMatchSnapshot();

  const codeFloat = gen(0.3);
  expect(codeFloat).toMatchSnapshot();
});

it('boolean', () => {
  const code = gen(true);
  expect(code).toMatchSnapshot();
});

it('empty array', () => {
  const code = gen([]);
  expect(code).toMatchSnapshot();
});

it('array of strings', () => {
  const code = gen(['string', 'data']);
  expect(code).toMatchSnapshot();
});

it('array of strings and numbers', () => {
  const code = gen(['string', 1]);
  expect(code).toMatchSnapshot();
});

it('empty object', () => {
  const code = gen({});
  expect(code).toMatchSnapshot();
});

it('nested objects', () => {
  const code = gen({ a: { b: { c: 'string' } }});
  expect(code).toMatchSnapshot();
});
