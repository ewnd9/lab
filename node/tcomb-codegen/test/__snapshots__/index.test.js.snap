exports[`test array of strings 1`] = `"t.list(t.String);"`;

exports[`test array of strings and numbers 1`] = `"t.list(t.union([t.String, t.Number]));"`;

exports[`test boolean 1`] = `"t.Boolean;"`;

exports[`test empty array 1`] = `"t.list(t.Any);"`;

exports[`test empty object 1`] = `"t.struct({});"`;

exports[`test nested objects 1`] = `
"t.struct({
  a: t.struct({
    b: t.struct({
      c: t.String
    })
  })
});"
`;

exports[`test number 1`] = `"t.Number;"`;

exports[`test number 2`] = `"t.Number;"`;

exports[`test string 1`] = `"t.String;"`;
